.github/workflows/build_and_test.yml

```
@@ -494,6 +495,41 @@ jobs:
         name: test-results-sparkr--8-${{ inputs.hadoop }}-hive2.3
         path: "**/target/test-reports/*.xml"
 
+  proto-breaking-changes-check:
+    needs: [precondition]
+    if: always() && fromJson(needs.precondition.outputs.required).proto-breaking-changes-check == 'true'
+    name: Spark Connect Proto Backwards Compat Check
```

nit but I would name it without capitalizing to be consistent with others. e.g.,

```suggestion
    name: Check Protobuf backward compatibility
```